torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -893,8 +893,11 @@ def render(  # type: ignore[override]
 
         assert len(self.input_nodes) >= 2 and self.output_node is not None
         X, W = self.input_nodes[0], self.input_nodes[1]
-        assert isinstance(X.layout, FixedLayout), "X.layout is not fixed"
-        assert isinstance(W.layout, FixedLayout), "W.layout is not fixed"
+        try:
+            assert isinstance(X.layout, FixedLayout), "X.layout is not fixed"
+            assert isinstance(W.layout, FixedLayout), "W.layout is not fixed"
+        except AssertionError:
+            raise NotImplementedError
```

Comment:
This seems less informative than the assertion error above? Is there a reason we need to raise `NotImplementedError` specifically?

Reply:
Agreed, I will try SkyLion's approach.

And good q, because `maybe_append_choice` requires `NotImplementedError` when choosing not to add a choice.

https://github.com/pytorch/pytorch/blob/34e420519df9c711ac997888470022cd7c4a90f6/torch/_inductor/codegen/common.py#L2687-L2700